src/main/java/org/jboss/threads/EnhancedQueueExecutorBase0.java

```diff
@@ -31,10 +31,10 @@
      * Padding fields.
      */
     @SuppressWarnings("unused")
-    int p00, p01, p02, p03,
+    long p00, p01, p02, p03,
         p04, p05, p06, p07,
         p08, p09, p0A, p0B,
-        p0C, p0D, p0E, p0F;
+        p0C, p0D, p0E;
```

Comment:
was `p0F` removal intentional? I think this is only 120 bytes

Reply:
org.jboss.threads.EnhancedQueueExecutor object internals iwth a 64 bit COOPS 8 bytes aligned instance
```
 OFFSET  SIZE                       TYPE DESCRIPTION                                  VALUE
      0    12                            (object header)                              N/A
     12     4                            (alignment/padding gap)                     
     16     8                       long EnhancedQueueExecutorBase0.p00               N/A
     24     8                       long EnhancedQueueExecutorBase0.p01               N/A
     32     8                       long EnhancedQueueExecutorBase0.p02               N/A
     40     8                       long EnhancedQueueExecutorBase0.p03               N/A
     48     8                       long EnhancedQueueExecutorBase0.p04               N/A
     56     8                       long EnhancedQueueExecutorBase0.p05               N/A
     64     8                       long EnhancedQueueExecutorBase0.p06               N/A
     72     8                       long EnhancedQueueExecutorBase0.p07               N/A
     80     8                       long EnhancedQueueExecutorBase0.p08               N/A
     88     8                       long EnhancedQueueExecutorBase0.p09               N/A
     96     8                       long EnhancedQueueExecutorBase0.p0A               N/A
    104     8                       long EnhancedQueueExecutorBase0.p0B               N/A
    112     8                       long EnhancedQueueExecutorBase0.p0C               N/A
    120     8                       long EnhancedQueueExecutorBase0.p0D               N/A
    128     8                       long EnhancedQueueExecutorBase0.p0E               N/A
    136     4                        int EnhancedQueueExecutorBase1.tailLock          N/A
    140     4                   TaskNode EnhancedQueueExecutorBase1.tail              N/A
    144     8                       long EnhancedQueueExecutorBase2.p00               N/A
    152     8                       long EnhancedQueueExecutorBase2.p01               N/A
    160     8                       long EnhancedQueueExecutorBase2.p02               N/A
    168     8                       long EnhancedQueueExecutorBase2.p03               N/A
    176     8                       long EnhancedQueueExecutorBase2.p04               N/A
    184     8                       long EnhancedQueueExecutorBase2.p05               N/A
    192     8                       long EnhancedQueueExecutorBase2.p06               N/A
    200     8                       long EnhancedQueueExecutorBase2.p07               N/A
    208     8                       long EnhancedQueueExecutorBase2.p08               N/A
    216     8                       long EnhancedQueueExecutorBase2.p09               N/A
    224     8                       long EnhancedQueueExecutorBase2.p0A               N/A
    232     8                       long EnhancedQueueExecutorBase2.p0B               N/A
    240     8                       long EnhancedQueueExecutorBase2.p0C               N/A
    248     8                       long EnhancedQueueExecutorBase2.p0D               N/A
    256     8                       long EnhancedQueueExecutorBase2.p0E               N/A
    264     8                       long EnhancedQueueExecutorBase2.p0F               N/A
    272     4                   TaskNode EnhancedQueueExecutorBase3.head              N/A
    276     4                            (alignment/padding gap)                     
    280     8                       long EnhancedQueueExecutorBase4.p00               N/A
    288     8                       long EnhancedQueueExecutorBase4.p01               N/A
    296     8                       long EnhancedQueueExecutorBase4.p02               N/A
    304     8                       long EnhancedQueueExecutorBase4.p03               N/A
    312     8                       long EnhancedQueueExecutorBase4.p04               N/A
    320     8                       long EnhancedQueueExecutorBase4.p05               N/A
    328     8                       long EnhancedQueueExecutorBase4.p06               N/A
    336     8                       long EnhancedQueueExecutorBase4.p07               N/A
    344     8                       long EnhancedQueueExecutorBase4.p08               N/A
    352     8                       long EnhancedQueueExecutorBase4.p09               N/A
    360     8                       long EnhancedQueueExecutorBase4.p0A               N/A
    368     8                       long EnhancedQueueExecutorBase4.p0B               N/A
    376     8                       long EnhancedQueueExecutorBase4.p0C               N/A
    384     8                       long EnhancedQueueExecutorBase4.p0D               N/A
    392     8                       long EnhancedQueueExecutorBase4.p0E               N/A
    400     8                       long EnhancedQueueExecutorBase4.p0F               N/A
    408     8                       long EnhancedQueueExecutorBase5.threadStatus      N/A
    416     8                       long EnhancedQueueExecutorBase6.p00               N/A
    424     8                       long EnhancedQueueExecutorBase6.p01               N/A
    432     8                       long EnhancedQueueExecutorBase6.p02               N/A
    440     8                       long EnhancedQueueExecutorBase6.p03               N/A
    448     8                       long EnhancedQueueExecutorBase6.p04               N/A
    456     8                       long EnhancedQueueExecutorBase6.p05               N/A
    464     8                       long EnhancedQueueExecutorBase6.p06               N/A
    472     8                       long EnhancedQueueExecutorBase6.p07               N/A
    480     8                       long EnhancedQueueExecutorBase6.p08               N/A
    488     8                       long EnhancedQueueExecutorBase6.p09               N/A
    496     8                       long EnhancedQueueExecutorBase6.p0A               N/A
    504     8                       long EnhancedQueueExecutorBase6.p0B               N/A
    512     8                       long EnhancedQueueExecutorBase6.p0C               N/A
    520     8                       long EnhancedQueueExecutorBase6.p0D               N/A
    528     8                       long EnhancedQueueExecutorBase6.p0E               N/A
    536     8                       long EnhancedQueueExecutorBase6.p0F               N/A
    544     8                       long EnhancedQueueExecutor.queueSize              N/A
    552     8                       long EnhancedQueueExecutor.timeoutNanos           N/A
    560     4                      float EnhancedQueueExecutor.growthResistance       N/A
    564     4                        int EnhancedQueueExecutor.peakThreadCount        N/A
    568     4                        int EnhancedQueueExecutor.peakQueueSize          N/A
    572     4                        int EnhancedQueueExecutor.activeCount            N/A
    576     4              ThreadFactory EnhancedQueueExecutor.threadFactory          N/A
    580     4                Set<Thread> EnhancedQueueExecutor.runningThreads         N/A
    584     4                 MXBeanImpl EnhancedQueueExecutor.mxBean                 N/A
    588     4                     Object EnhancedQueueExecutor.handle                 N/A
    592     4       AccessControlContext EnhancedQueueExecutor.acc                    N/A
    596     4                     Waiter EnhancedQueueExecutor.terminationWaiters     N/A
    600     4                   Executor EnhancedQueueExecutor.handoffExecutor        N/A
    604     4   UncaughtExceptionHandler EnhancedQueueExecutor.exceptionHandler       N/A
    608     4                   Runnable EnhancedQueueExecutor.terminationTask        N/A
    612     4                  LongAdder EnhancedQueueExecutor.submittedTaskCounter   N/A
    616     4                  LongAdder EnhancedQueueExecutor.completedTaskCounter   N/A
    620     4                  LongAdder EnhancedQueueExecutor.rejectedTaskCounter    N/A
    624     4                  LongAdder EnhancedQueueExecutor.spinMisses             N/A
    628     4                            (loss due to the next object alignment)
Instance size: 632 bytes
Space losses: 8 bytes internal + 4 bytes external = 12 bytes total
```
You're right I can left the original version too  to be safe: I've changed this because if someone will access the last 4 bytes of the object header (ie at offset 8), it can being at most 1 cache lines (8 + 128 = 136) leaving `tailLock` untouched.